src/Stack/Package.hs

```diff
@@ -734,14 +734,14 @@ libraryFiles lib = do
         resolveFilesAndDeps
             Nothing
             (dirs ++ [dir])
-            (names <> exposed)
+            names
```

Comment:
This line alone would be a sufficient fix (see `names = bnames ++ exposed`!) but I felt that `Set` is the better datastructure than `[]` in this case. Order ~~isn't~~ doesn't seem to be important and we don't want any duplicates.

I actually think that we should use `Set` instead of `[]` much more pervasively to prevent ourselves from wondering whether duplicates and order actually matter in each usecase.